### ISSUE TITLE
Bug: Menu Grouping Semantics

### ISSUE THREAD

**Issue by sampotts** [videojs/v10#1639]
Currently in the menus, we only have a `role="group"` and a label prop that set `aria-label` but it was unused. We also have a `Label` component but it's not associated with the group. We should:

- Remove the label prop and pass through `aria-label` and `aria-labelledby` instead.
- Rename `Label` to `GroupLabel` which also aligns closer to Base UIs pattern.
- Register the `GroupLabel` with the parent group and set the `aria-labelledby` prop on the group.
